Docs 6342 NetSuite Troubleshooting Topic (#421)
mulesoft/docs-connectors: modules/ROOT/pages/netsuite/netsuite-troubleshooting.adoc

= Netsuite Connector Troubleshooting - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: anypoint studio, esb, connectors, http, https, http headers, troubleshooting, rest, raml

You might encounter the following issues when working with the Anypoint Connector for NetSuite (NetSuite Connector):

* <<connectivity-test-failure, A connectivity test fails during application startup.>>
* <<logging-request-responses, The connector does not log requests and responses.>>
* <<permissions-error, The connector returns a permissions error when fetching metadata.>>

== Connectivity Test Failure During Application Startup [[connectivity-test-failure]]

> If a connectivity test fails during application startup, and you see the following statement in the stacktrace, configure a reconnection strategy for the connector:

Reviewer: ```suggestion
If a connectivity test fails during application startup and you see the following statement in the stack trace, configure a reconnection strategy for the connector:
```

[source,xml,linenums]
----
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998) ~[?:1.8.0_221]
----

> To configure a reconnection strategy for the connector in Anypoint Studio (Studio), add these lines to the *Configure XML* tab in the connector properties window :

Reviewer: ```suggestion
To configure a reconnection strategy for the connector in Anypoint Studio (Studio), add these lines to the *Configure XML* tab in the connector properties window:
```

[source,xml,linenums]
----
<reconnection failsDeployment="true">
	<reconnect count="2" frequency="5" />
</reconnection>
----

For example, the configuration XML for the connector might look like this after you add the reconnection strategy:

[source,xml,linenums]
----
<netsuite:config name="NetSuite_Config" doc:name="NetSuite Config" doc:id="8a4f73c2-e5e1-4f5d-8978-980069828ff0" >
	<netsuite:login-authentication-connection email="${netsuite.email}" password="${netsuite.password}" account="${netsuite.account}" roleId="${netsuite.roleId}" applicationId="${netsuite.applicationId}" readTimeout="60000" connectionTimeout="60000">
		<reconnection failsDeployment="true">
			<reconnect count="2" frequency="5" />
		</reconnection>
	</netsuite:login-authentication-connection>
----

In Studio, you can also use the *Advanced* tab in the connector properties window to configure a reconnection strategy. To do so, complete the following fields:

[%header%autowidth.spread]
|===
|Field |Value
|Reconnection strategy |`Standard`
|Frequency (ms) |`5`
|Reconnection Attempts |`2`
|===

== No Logging of Log Requests and Responses [[logging-request-responses]]

> To log requests and responses when using the NetSuite connector, add this XML statement to the `Log4j` configuration file:

Reviewer: ```suggestion
To log requests and responses when using NetSuite Connector, add this XML statement to the `Log4j` configuration file:
```
OR
```suggestion
To log requests and responses when using the connector, add this XML statement to the `Log4j` configuration file:
```

[source,xml,linenums]
----
<AsyncLogger
  name="org.mule.module.netsuite.extension.internal.connection.HttpWithCookieMessageDispatcher"
  level="DEBUG"/>
<Root level="debug">
    <AppenderRef ref="Console" />
</Root>
----

> == Permissions Error when Fetching Metadata [[permissions-error]]

Reviewer: ```suggestion
== Permissions Error When Fetching Metadata [[permissions-error]]
```
